### PR TITLE
Next consistent cell value conversion from str

### PR DESCRIPTION
**Pull request by ddimaria**
- [x] Ensure that most &str-to-CellValue conversions utilize the `grid_controller::set_cell_value_operations()`
- [x] Fix all the broken things in Next

### CODE REVIEW

File: quadratic-core/src/values/convert.rs [hunk 1]
```diff
@@ -29,18 +29,30 @@ where
 }
 impl From<String> for CellValue {
     fn from(value: String) -> Self {
-        CellValue::Text(value)
+        value.as_str().into()
     }
 }
+// TODO(ddimaria): implement Duration and Instant
 impl From<&str> for CellValue {
```
Comment (davidkircos) [inline quadratic-core/src/values/convert.rs:36]: Andrew "this should just convert to a CellValue::Text()".

File: quadratic-core/src/controller/cells.rs [hunk 2]
```diff
@@ -26,13 +26,22 @@ impl GridController {
         value: String,
         cursor: Option<String>,
     ) -> TransactionSummary {
+        let ops = self.set_cell_value_operations(sheet_id, pos, &value);
+        self.transact_forward(ops, cursor).await
+    }
+    pub fn set_cell_value_operations(
```
Comment (davidkircos) [inline quadratic-core/src/controller/cells.rs:32]: Use this in Paste/ImportCSV/User inputting a cell value/etc.

File: quadratic-core/src/values/convert.rs [hunk 3]
```diff
@@ -29,18 +29,30 @@ where
...
     fn from(value: &str) -> Self {
-        CellValue::Text(value.to_string())
+        let parsed = Self::strip_percentage(Self::strip_currency(value)).trim();
```
Comment (ddimaria) [inline quadratic-core/src/values/convert.rs:38]: I'm moving this fn to `tesst_util.rs`

File: quadratic-core/src/grid/file/v1_4.rs [hunk 4]
```diff
@@ -159,15 +158,9 @@ pub enum Any {
...
-            Any::Number(n) => match BigDecimal::from_f64(n) {
-                Some(n) => CellValue::Number(n),
-                None => CellValue::Text(n.to_string()),
-            },
-            Any::String(s) => match BigDecimal::from_str(&s) {
-                Ok(n) => CellValue::Number(n),
-                Err(_) => CellValue::Text(s),
-            },
-            Any::Boolean(b) => CellValue::Logical(b),
+            Any::Number(n) => n.into(),
```
Comment (ddimaria) [inline quadratic-core/src/grid/file/v1_4.rs:161]: @davidkircos did you want me to revert this change?

File: quadratic-core/src/wasm_bindings/mod.rs [hunk 5]
```diff
@@ -230,23 +229,9 @@ impl JsGridProxy {
 #[async_trait(?Send)]
 impl GridProxy for JsGridProxy {
     async fn get(&mut self, pos: Pos) -> CellValue {
-        let jsvalue = match self.get_cell_jsvalue(pos).await {
```
Comment (ddimaria) [inline quadratic-core/src/wasm_bindings/mod.rs:233]: @davidkircos did you want me to revert this one?  This seems like it should use the new conversions, but you have more familiarity with the functionality.

File: quadratic-core/src/controller/cells.rs (hunk 2 above)
Comment (ddimaria) [reply quadratic-core/src/controller/cells.rs:32]: Paste and import are implemented.  Do you know where I can find "User inputting a cell value/etc." @davidkircos ?

**Review by davidfig — APPROVED** [review]

File: quadratic-core/src/values/convert.rs (hunk 1 above)
Comment (ddimaria) [reply quadratic-core/src/values/convert.rs:34]: This has been updated

File: quadratic-core/src/wasm_bindings/mod.rs (hunk 5 above)
Comment (ddimaria) [reply quadratic-core/src/wasm_bindings/mod.rs:234]: I'm reverting this and I can always add this back.
